### ISSUE TITLE
Hydrated: HiveError: Cannot write, unknown type: Plan. Did you forget to register an adapter? 

### ISSUE THREAD

**Issue by orsenkucher** [felangel/bloc#1453]
**Describe the bug**
Previously we used `json.encode` to encode our states:
```
If value contains objects that are not directly encodable to a JSON
string (a value that is not a number, boolean, string, null, list or a map
with string keys), the [toEncodable] function is used to convert it to an
object that must be directly encodable.

If [toEncodable] is omitted, it defaults to a function that returns the
result of calling .toJson() on the unencodable object.
```
But Hive's serialization system doesn't have this nice property of calling `.toJson` on values it can't serialize directly.

**To Reproduce**
- Have a list of custom types (even with toJson method defined)
- Have nested `pkg:freezed` types

**Expected behavior**
Everything goes smooth

**Comment by orsenkucher** [comment]
- Maybe we can run `json.encode` on our state before sending it to hive. But then `Map<int, dynamic>` will panic((
- Or search for `.toJson` method ourselves and substitute value with returned value

Not rly sure on either of this

**Comment by felangel** [comment]
Fixed in #1460 and published in v6.0.0-dev.2 of `hydrated_bloc` 🎉
